Spigot-Server-Patches/0393-Add-more-Witch-API.patch

```diff
@@ -0,0 +1,179 @@
+     void c(AttributeModifier var1);
+ 
+     void b(UUID var1);
+diff --git a/src/main/java/net/minecraft/server/EntityMonster.java b/src/main/java/net/minecraft/server/EntityMonster.java
+index 7cba60e47..f6231f3e5 100644
+--- a/src/main/java/net/minecraft/server/EntityMonster.java
++++ b/src/main/java/net/minecraft/server/EntityMonster.java
+@@ -8,6 +8,7 @@ public abstract class EntityMonster extends EntityCreature implements IMonster {
+         this.b_ = 5;
+     }
+ 
++    public SoundCategory getSoundCategory() { return bV(); } // Paper - OBFHELPER
+     public SoundCategory bV() {
+         return SoundCategory.HOSTILE;
+     }
+diff --git a/src/main/java/net/minecraft/server/EntityWitch.java b/src/main/java/net/minecraft/server/EntityWitch.java
+index b5f676f04..41673ec9a 100644
+--- a/src/main/java/net/minecraft/server/EntityWitch.java
++++ b/src/main/java/net/minecraft/server/EntityWitch.java
+@@ -1,5 +1,11 @@
+ package net.minecraft.server;
+ 
++// Paper start
++import com.destroystokyo.paper.event.entity.WitchReadyPotionEvent;
++import org.bukkit.craftbukkit.inventory.CraftItemStack;
++import org.bukkit.entity.Witch;
++// Paper end
++
+ import java.util.Iterator;
+ import java.util.List;
+ import java.util.UUID;
+@@ -8,9 +14,9 @@ import javax.annotation.Nullable;
+ public class EntityWitch extends EntityMonster implements IRangedEntity {
+ 
+     private static final UUID a = UUID.fromString("5CD17E52-A79A-43D3-A529-90FDE04B181E");
+-    private static final AttributeModifier b = (new AttributeModifier(EntityWitch.a, "Drinking speed penalty", -0.25D, 0)).a(false);
++    private static final AttributeModifier b = (new AttributeModifier(EntityWitch.a, "Drinking speed penalty", -0.25D, 0)).a(false); private static final AttributeModifier DRINKING_SPEED = b; // Paper - OBFHELPER
+     private static final DataWatcherObject<Boolean> c = DataWatcher.a(EntityWitch.class, DataWatcherRegistry.i);
+-    private int bC;
++    private int bC; public int getPotionUseTimeLeft() { return bC; } public void setPotionUseTimeLeft(int timeLeft) { bC = timeLeft; } // Paper - OBFHELPER
+ 
+     public EntityWitch(World world) {
+         super(EntityTypes.WITCH, world);
+@@ -44,10 +50,12 @@ public class EntityWitch extends EntityMonster implements IRangedEntity {
+         return SoundEffects.ENTITY_WITCH_DEATH;
+     }
+ 
++    public void setDrinkingPotion(boolean drinkingPotion) { a(drinkingPotion); } // Paper - OBFHELPER
+     public void a(boolean flag) {
+         this.getDataWatcher().set(EntityWitch.c, Boolean.valueOf(flag));
+     }
+ 
++    public boolean isDrinkingPotion() { return l(); } // Paper - OBFHELPER
+     public boolean l() {
+         return ((Boolean) this.getDataWatcher().get(EntityWitch.c)).booleanValue();
+     }
+@@ -100,18 +108,17 @@ public class EntityWitch extends EntityMonster implements IRangedEntity {
+                 }
+ 
+                 if (potionregistry != null) {
+-                    // Paper start
```

Comment:
maybe its less clear in patch form, but could we maybe just split the method into 3 here instead, so we don't have to move all the logic out? then lot of the obf helpers wouldnt even be needed.